### ISSUE TITLE
GCP persistent disk device name is 'sdb', not 'pkb-xxxxxxxx-0-data-0-0'

### ISSUE THREAD

**Issue by HarpPDX** [GoogleCloudPlatform/PerfKitBenchmarker#2003]
This appears to be an issue with the GCP API, but not sure how/where to report that so raising the issue here as PKB users may run into the issue.

The error from PKB looks like this:
```
2019-09-26 08:54:31,060 406c8cfa Thread-42 mongodb_stress(1/1) INFO     Running: ssh -A -p 22 perfkit@35.225.149.231 -2 -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o IdentitiesOnly=yes -o PreferredAuthentications=publickey -o PasswordAuthentication=no -o ConnectTimeout=5 -o GSSAPIAuthentication=no -o ServerAliveInterval=30 -o ServerAliveCountMax=10 -i /tmp/perfkitbenchmarker/runs/406c8cfa/perfkitbenchmarker_keyfile -o ControlPath="/tmp/perfkitbenchmarker/runs/406c8cfa/ssh/%C" -o ControlMaster=auto -o ControlPersist=30m [[ -d /mnt ]] && sudo umount /mnt; sudo mke2fs -F -E lazy_itable_init=0,discard -O ^has_journal -t ext4 -b 4096 /dev/pkb-406c8cfa-0-data-0-0
2019-09-26 08:54:31,299 406c8cfa Thread-42 mongodb_stress(1/1) INFO     Ran: {ssh -A -p 22 perfkit@35.225.149.231 -2 -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o IdentitiesOnly=yes -o PreferredAuthentications=publickey -o PasswordAuthentication=no -o ConnectTimeout=5 -o GSSAPIAuthentication=no -o ServerAliveInterval=30 -o ServerAliveCountMax=10 -i /tmp/perfkitbenchmarker/runs/406c8cfa/perfkitbenchmarker_keyfile -o ControlPath="/tmp/perfkitbenchmarker/runs/406c8cfa/ssh/%C" -o ControlMaster=auto -o ControlPersist=30m [[ -d /mnt ]] && sudo umount /mnt; sudo mke2fs -F -E lazy_itable_init=0,discard -O ^has_journal -t ext4 -b 4096 /dev/pkb-406c8cfa-0-data-0-0}  ReturnCode:1
STDOUT:
STDERR: umount: /mnt: not mounted.
mke2fs 1.44.1 (24-Mar-2018)
The file /dev/pkb-406c8cfa-0-data-0-0 does not exist and no size was specified.
```
It appears the PKB is calling the CLI with the correct parameter to name the device:
```
2019-09-26 08:51:51,421 406c8cfa Thread-42 mongodb_stress(1/1) vm_util.py:338 INFO     Running: gcloud compute instances attach-disk pkb-406c8cfa-0 --format json --quiet --project ssg-benchmarks --zone us-central1-c --device-name pkb-406c8cfa-0-data-0-0 --disk pkb-406c8cfa-0-data-0-0
```

When I log in to the VM, I can see (via lsblk) that the device name is 'sdb' not pkb-xxxxxxxx-0-data-0-0.

Note that I see the same problem when creating a VM and attaching a disk through the console/GUI.

**Comment by dlott** [comment]
Hi Jason, 

Can you share your full pkb command line and/or log file?

Thanks,
David

**Comment by HarpPDX** [comment]
Complete log file attached.  Search for 'pkb-406c8cfa-1-data-0-0' to get to the interesting stuff.
[pkb_406c8cfa.log](https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/files/3682720/pkb_406c8cfa.log)

Note that this occurs with pd-ssd and pd-standard disk types.  Local disks work fine.

**Comment by dlott** [comment]
What modifications do you have for `PerfKitBenchmarker version: v1.3.0.intel-54-geb9ebb4e`?  Is there a standard benchmark that also exhibits the issue (i.e. not mongodb_stress nor ubuntu1904)?  I think you might be able to repro it with cluster_boot and picking the proper --image?

**Comment by s-deitz** [comment]
If I'm reading the log correctly, it looks like there are three VMs created and we try to attach the disk to the two worker VMs. I see vm_count set to 1 above though. Maybe that isn't plumbed through to the benchmark setup.

**Comment by HarpPDX** [comment]
Fair enough...let me attempt to reproduce with the master branch from a fresh clone of this repo. I'll be back. ;-)

**Comment by HarpPDX** [comment]
Alright...I was able to reproduce the problem on the latest code from this repo. At the same time, I was able to figure out why it is happening -- I have/had the gce_ssd_interface flag set to NVME. I was under the impression that this flag only applies to "local" drives, but the GetDevicePath function in gce_disk.py uses this flag regardless of the drive type.  The function should probably check the drive type before using the flag.
```
  def GetDevicePath(self):
    """Returns the path to the device inside the VM."""
    if FLAGS.gce_ssd_interface == SCSI:
      return '/dev/disk/by-id/google-%s' % self.name
    elif FLAGS.gce_ssd_interface == NVME:
      return '/dev/%s' % self.name
```

**Comment by dlott** [comment]
Thanks for the update.  It seems like there's a workaround of just using the default value for `--gce_ssd_interface`?  If so we'll add this to the backlog.

**Comment by HarpPDX** [comment]
@dlott Yes, that is the workaround.  Thanks.

**Comment by darabearr** [comment]
Resolved in commit 9915b334621acc8c262fb47ff38846a873d54531
